### DISCUSSION TITLE
CMS for JupyterBook?

### DISCUSSION THREAD

**Discussion by maximikos** [executablebooks/meta#1123] [Q&A]
**TL;DR**: I want to link a published JupyterBook to a content management system (CMS) and have tried so with [DecapCMS](https://decapcms.org). While new blog-like entries can be generated, rebuilding the book results in overwriting essential parts of the CMS configuration. Now I'm wondering if such a JupyterBook + CMS link can be achieved at all, and if so, how? And if not, can you recommend other tools/setups for our purpose?

---

Hi,

I've had great experiences with using JupyterBooks (and the greater ExecutableBooks framework) for my own work and was planning on using it for another project. As part of increasing outreach activities on environmental systems analysis, some members of our scientific society want to create a community-curated online textbook. We imagine it as a platform where verified society members can create/edit/review material which is to be made freely accessible to the greater public (for now focusing on high-school students, in case you are wondering).

If we were only, say, 5 people with some coding experience, it would of course be relatively straight-forward to co-produce such a JupyterBook. However, for us it's a little more complicated. The tricky part is to: 1) make editing easy, even for folks that have never written a single line of code and have no clue what GitHub is, and 2) allow authorised access to multiple (>>5) people, 3) who may have different roles (e.g. authoring a new article vs reviewing/validating said article). After looking all over the place, I got the impression that linking a CMS to a JupyterBook could solve at least 1), and depending on its flexibility perhaps also 2) and 3).

We have a placeholder JupyterBook (all pretty standard) that we deploy via Netlify. That made it natural to look into DecapCMS (formerly Netlify CMS). Following their [guide](https://decapcms.org/docs/add-to-your-site/), I managed to access the CMS on the published JupyterBook and even create a new entry. However, when re-deploying the JupyterBook, the sections necessary for configuring the CMS are removed/overwritten. That is, the to-be-created `_build/html/admin` folder is removed and to-be-added lines in the `_build/docs/index.html` file are overwritten. Changes via CMS to an existing markdown file or newly created markdown files remain.

Based on this outcome, I have a few questions:
- Is it possible at all to link a JupyterBook to a CMS?
- Has anyone experience with that and could perhaps provide a working example? (Is @curvenote one?)
- Would it be possible to circumvent the overwriting of sections via some GitHub Actions? (I have no experience with that at all...)
- If such a link is not possible, can you recommend alternatives (tools, workflows etc.)?

And just in case you are wondering: We would prefer to stick to JupyterBooks as: 1) we want to include interactive visualisations and code examples, and 2) have limited resources (as of now, our work is completely voluntary and we want it to be freely accessible as well).

Many thanks in advance for any input you might have!
